sdk/digitaltwins/azure-digitaltwins-core/API design.md

```diff
@@ -771,12 +771,55 @@ public DigitalTwinsResponse<Void> updateComponentWithResponse(String digitalTwin
 </details>
 
 ## Query
-<details>
-<summary><b>APIs</b></summary>
+<details><summary><b>Async APIs</b></summary>
+
+  ```java
+  /**
+  * Query digital twins.
+  *
+  * @param query The query string, in SQL-like syntax.
+  * @return A {@link PagedFlux} of application/json for the query result items.
+  */
+  @ServiceMethod(returns = ReturnType.COLLECTION)
+  public PagedFlux<String> query(String query)
+
+  /**
+  * Query digital twins.
+  *
+  * @param query The query string, in SQL-like syntax.
+  * @param clazz The model class to convert the query response to.
+  * @param <T> The generic type to convert the query response to.
+  * @return A {@link PagedFlux} of application/json for the query result items.
+  */
+  @ServiceMethod(returns = ReturnType.COLLECTION)
+  public <T> PagedFlux<T> query(String query, Class<T> clazz)
```

Comment:
query as object?

Reply:
This did not come up in yesterday's discussion. Do we really need query as object? Is <T> not enough?

Reply:
Resolved in stand up